### ISSUE TITLE
hddtemp deprecated, breaking plugin

### ISSUE THREAD

**Issue by popey** [influxdata/telegraf#11738]
### Relevant telegraf.conf

```toml
[[inputs.hddtemp]]
```

### Logs from Telegraf

```text
Aug 29 20:43:20 earth telegraf[923]: 2022-08-29T19:43:20Z E! [inputs.hddtemp] Error in plugin: dial tcp 127.0.0.1:7634: connect: connection refused                                          
Aug 29 20:43:30 earth telegraf[923]: 2022-08-29T19:43:30Z E! [inputs.hddtemp] Error in plugin: dial tcp 127.0.0.1:7634: connect: connection refused                                          
Aug 29 20:43:40 earth telegraf[923]: 2022-08-29T19:43:40Z E! [inputs.hddtemp] Error in plugin: dial tcp 127.0.0.1:7634: connect: connection refused
```

### System info

1.23.4 on Ubuntu 22.04 LTS

### Steps to reproduce

1. Install Telegraf on Ubuntu 20.04
2. Install `hddtemp` and configure 
3. Enable `input.hddtemp` in telegraf configuration
4. Upgrade to Ubuntu 22.04
5. Observe error in logs as `hddtemp` package is removed
...

### Expected behavior

Given this is an Ubuntu packaging problem, I'm not expecting it to continue working. 

### Actual behavior

Temps are not gathered.

### Additional info

Some research suggests there's a new API for gathering drive temperatures. Perhaps we need a new plugin which introspects this API.

**Comment by reimda** [comment]
Telegraf's plugin inputs/smart gathers temperature (in the measurement `smart_device` field `temp_c`) in addition to lots more information about drives. It looks like the hddtemp daemon also queries SMART to get the temperature, so using inputs/smart should provide the same data.

It looks like the hddtemp project has not been updated for many years. The last code change I see is 2007.

Given that the hddtemp project seems dead and that there is already a workaround, I wonder if we should deprecate the hddtemp plugin. We could add a note in its readme directing people to migrate to inputs/smart.

**Comment by chrishoage** [comment]
> migrate to inputs/smart.

Would be nice if this was replaced with `hdparm -H`

> Read the temperature from some (mostly Hitachi) drives. Also reports if the temperature is within operating condition range (this may not be reliable). Does not cause the drive to spin up if idle.

https://man7.org/linux/man-pages/man8/hdparm.8.html

While only Hitachi/HGST/WD drives support this those cover a large amount of drives, and not needing the drive to be spun up is very valuable to be able to monitor temperatures while allowing the spindown of drives for power or noise reasons. 

Example output

```
❯ sudo hdparm -H $ALL_HDD_DEV

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEJLL6NS:
 drive temperature (celsius) is:  28
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEK49SBS:
 drive temperature (celsius) is:  30
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEKDNZ0S:
 drive temperature (celsius) is:  26
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEKDXVTS:
 drive temperature (celsius) is:  26
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK2334PEJM9B3T:
 drive temperature (celsius) is:  28
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK2334PEK4AXTT:
 drive temperature (celsius) is:  26
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_WDC_WD140EDGZ-11_9LK4YMLG:
 drive temperature (celsius) is:  30
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_WDC_WD140EDGZ-11_9MHV5A5U:
 drive temperature (celsius) is:  30
 drive temperature in range:  yes

/dev/disk/by-id/scsi-SATA_WDC_WD140EDGZ-11_9MJ2NDKU:
 drive temperature (celsius) is:  25
 drive temperature in range:  yes
```

```
❯ sudo hdparm -C $ALL_HDD_DEV

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEJLL6NS:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEK49SBS:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEKDNZ0S:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK1334PEKDXVTS:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK2334PEJM9B3T:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_HGST_HDN724040AL_PK2334PEK4AXTT:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_WDC_WD140EDGZ-11_9LK4YMLG:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_WDC_WD140EDGZ-11_9MHV5A5U:
 drive state is:  standby

/dev/disk/by-id/scsi-SATA_WDC_WD140EDGZ-11_9MJ2NDKU:
 drive state is:  standby
```

**Comment by ThomDietrich** [comment]
Did anyone find an appropriate workaround or solution for this issue? Cheers!

**Comment by srebhan** [comment]
@ThomDietrich you might want to try the [`sensors` input plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/sensors)... Does that work?

**Comment by ThomDietrich** [comment]
@srebhan Thanks for the response.
`sensors` works for some systems, but not all. Also, I found `sensors` provides less useful contextual information regarding harddisks:

hddtemp:

![image](https://user-images.githubusercontent.com/2870104/235915837-452e2860-2922-4be2-b271-631a65673b2d.png)

sensors:

![image](https://user-images.githubusercontent.com/2870104/235916591-637af0e5-d84b-4c5f-a2e5-8d4bb0697054.png)

**Comment by kirias** [comment]
Hi, 
> 2. Install hddtemp and configure
Have you set RUN_DAEMON="true" in /etc/default/hddtemp?
Then service hddtemp restart

**Comment by douginoz** [comment]
I'm running telegraf in docker and had this problem.  I resolved it by modifying my docker compose file to include:
```
services:
  telegraf:
    network_mode: host
```

Or via docker run:
`docker run -d --network=host `

I was working on the assumption that 127.0.0.1 inside docker isn't the same as the host, which turns out to be [true](https://www.howtogeek.com/devops/how-to-connect-to-localhost-within-a-docker-container/).

**Comment by porteusconf** [comment]
For getting temperature of SSD/HD, direct folks to newer[ `inputs.smartctl` ](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/smartctl) instead of older `inputs.smart` plugin. The readme explains the smartctl plugin parses json data from `smartctl --json`. In 2019, with v7.0, the new `--json` option first appeared. At that time, compared to other temperature attributes, [author Franke said ]( https://www.smartmontools.org/ticket/1164) 

> "it is recommended to use JSON field `temperature.current` instead. This also evaluates standardized ATA/SATA temperature sources (SCT Status, Device Statistics) **_and also works for SCSI/SAS and NVMe devices._**"

 (And  even some hardware raid, so just about any storage device!).   I see go code in`inputs.smartctl` that seems to parse`temperature.current` so that is your best bet. With the cavaet that while`--json` option is now stable and mature, there are always edge cases of problems getting even basic values like temperature, hours, etc for certain drives. (For insight, see detailed overview of recommended ways to parse `smartctl --json` in [Troubleshooting data inconsistencies issue 157 ](https://github.com/prometheus-community/smartctl_exporter/issues/157  ) 

TL/DR: use [inputs.smartctl](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/smartctl) instead of inputs.smart (assuming you can install smartmontools 7.3 or newer) and deprecate hddtemp. while providing this advice, and warning all new issues in hddtemp will likely be marked "won't fix" from now on.

**Comment by powersj** [comment]
@srebhan already on it ;)

@porteusconf thanks for the suggestion about smartctl. That looks like a valid way forward for users affected by this.

I went looking to see if we should deprecate the entire plugin, but it looks like hddtemp still exists in Fedora, Arch, and SuSE. That means this only has affected the Debian/Ubuntu ecosystem. With the smartctl the data can still get obtained.

Because I don't think we should be doing anything different in Telegraf specifically, I'll document this in the plugin and close this issue.
